code/modules/mob/living/simple_animal/hostile/space_dragon.dm

```diff
@@ -292,25 +320,31 @@
  * It creates a hotspot on the given turf, damages any living mob with 30 burn damage, and damages mechs by 50.
  * It can only hit any given target once.
  * Arguments:
- * * turf/T - The turf to trigger the effects on.
+ * * turf/turf - The turf to trigger the effects on.
+ * * affected_volume - The number of turfs that got influenced by the fire stream.
  */
-/mob/living/simple_animal/hostile/space_dragon/proc/dragon_fire_line(turf/T)
+/mob/living/simple_animal/hostile/space_dragon/proc/dragon_fire_line(turf/turf, affected_volume, total_plasma_released)
 	var/list/hit_list = list()
 	hit_list += src
-	new /obj/effect/hotspot(T)
-	T.hotspot_expose(700,50,1)
-	for(var/mob/living/L in T.contents)
-		if(L in hit_list)
+	if(isopenturf(turf))
+		var/turf/open/open_turf = turf
+		open_turf.atmos_spawn_air("plasma=[total_plasma_released / affected_volume];o2=[OXYGEN_RATIO * total_plasma_released / affected_volume];TEMP=[PLASMA_TEMPERATURE]")
+	new /obj/effect/hotspot(turf)
+	turf.hotspot_expose(exposed_temperature = PLASMA_TEMPERATURE, exposed_volume = 50, soh = 1)
+	for(var/mob/living/living in turf.contents)
+		if("carp" in living.faction)
+			hit_list += living
+		if(living in hit_list)
 			continue
-		hit_list += L
-		L.adjustFireLoss(30)
-		to_chat(L, span_userdanger("You're hit by [src]'s fire breath!"))
+		hit_list += living
+		living.apply_damages(brute = 10, burn = 20)
+		to_chat(living, span_userdanger("You're hit by [src]'s fire breath!"))
```

Comment:
Could be nice to have this be a short bubble alert? Maybe something like "burned by dragon!" or something? Not sure.